### PR TITLE
fix(metadata): fix ClearMetadata

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

ClearMetadata was not working as intended and test covarage needed improving

## Changeset

- Added the following methods to the INativeClient interface: ClearNativeMetadata, SetNativeMetadata, GetNativeMetadata to simplify the code and make it easier to understand the relationship between Native and c# metadata.
- Added new tests for ClearMetadata

## Testing

- Added new tests for ClearMetadata

### CODE REVIEW

File: features/android/android_metadata.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        And the event "metaData.test.test1" equals "test1"
+        And the event "metaData.test.test2" is null
+        And the event "metaData.test3.test3" equals "test3"
+        And the event "metaData.test4" is null
+
+
+
+
+
+
```
Comment (fractalwrench) [inline features/android/android_metadata.feature:20]:
```suggestion
        And the event "metaData.test4" is null
```

File: features/ios/ios_metadata.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        And the event "metaData.test.test2" is null
+        And the event "metaData.test3.test3" equals "test3"
+        And the event "metaData.test4" is null
+
+
+
+
+
+
+
```
Comment (fractalwrench) [inline features/ios/ios_metadata.feature:22]:
```suggestion

```

File: src/BugsnagUnity/Native/Android/NativeInterface.cs [hunk 3]
```diff
@@ -544,6 +553,8 @@ public void AddToTab(string tab, string key, string value)
                 new object[] { tab, key, value });
         }
 
+       
+
```
Comment (fractalwrench) [inline src/BugsnagUnity/Native/Android/NativeInterface.cs:557]:
```suggestion
```

File: src/BugsnagUnity/Native/Android/NativeClient.cs [hunk 4]
```diff
@@ -132,6 +112,28 @@ public LastRunInfo GetLastRunInfo()
...
+        {
+            NativeInterface.ClearMetadata(section,key);
+        }
+
+        public IDictionary<string, object> GetNativeMetadata()
+        {
+            return NativeInterface.GetMetadata();
+        }
+
+        public void SetNativeMetadata(string section, IDictionary<string, object> data)
```
Comment (fractalwrench) [inline src/BugsnagUnity/Native/Android/NativeClient.cs:130]: I'm not sure if this function makes sense - I would assume that this clears any existing metadata, then adds all the key-value pairs in `data`. Right now the implementation just adds metadata, so it feels like the values could be wrong if someone forgot to call `ClearNativeMetadata` first?

File: src/BugsnagUnity/Payload/Metadata.cs [hunk 5]
```diff
@@ -32,61 +33,45 @@ public void AddMetadata(string section, IDictionary<string, object> metadataSect
             }
             if (SectionExists(section))
             {
-
                 var existingSection = (IDictionary<string, object>)Get(section);
 
-                foreach (var entry in metadataSection)
+                foreach (var key in metadataSection.Keys.ToList())
```
Comment (fractalwrench) [inline src/BugsnagUnity/Payload/Metadata.cs:38]: Is there an advantage to using the keys directly rather than an entryset?

Comment (rich-bugsnag) [reply src/BugsnagUnity/Payload/Metadata.cs:38]: It allows for the set to be edited while iterating

File: src/BugsnagUnity/Native/Android/NativeClient.cs (hunk 4 above)
Comment (rich-bugsnag) [reply src/BugsnagUnity/Native/Android/NativeClient.cs]: The behaviour is correct, but the name was missleading, i have renamed it 👍

**Review by fractalwrench — APPROVED** [review]
LGTM

**Pull request merged**